test/SILOptimizer/pound_assert.swift

```diff
@@ -154,3 +177,27 @@ func test_CustomStruct() {
   #assert(cs.x.1 == 2)
   #assert(cs.y == 3)
 }
+
+//===----------------------------------------------------------------------===//
+// Mutation
+//===----------------------------------------------------------------------===//
+
+struct MutableStruct {
+  var x: (Int, Int)
+  var y: Int
+}
+
+func addOne(to target: inout Int) {
+  target += 1
+}
+
+func functionWithMutations(_ ms: MutableStruct) -> Int {
+  var myMs = ms
+  addOne(to: &myMs.x.0)
+  addOne(to: &myMs.y)
+  return myMs.x.0 + myMs.x.1 + myMs.y
```

Comment:
It would be good to make these test a bit more comprehensive. In particular, it would be good to make sure there is not unexpected aliasing (for example, between `myMs` and `ms` above).

It would also be good to test the case where you are storing an aggregate to a projection. Something like:

```
struct Foo {
  var x: Int 
  var y: Int

  init() {
   x = 7
   y = 8
  }
}

struct HasFoo {
  var f = Foo()
}

func foo() -> Bool {
  var h = HasFoo()
  var hp = h
  h.f.x = 22
  hp.f = h.f
  return (hp.f.x == 7)
}

func bar() {
  #assert(foo())
}
```

Reply:
I added a test for undesired aliasing, and a test for storing aggregate to projection.